fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/cursors/SizeStatisticsResultsTest.java

```diff
@@ -0,0 +1,298 @@
+        stats.updateStatistics(new KeyValue("ab".getBytes(), "cd".getBytes())); // total size 4
+        
+        // Size 8: bucket 3 (since 2^3 = 8)
+        stats.updateStatistics(new KeyValue("abcd".getBytes(), "efgh".getBytes())); // total size 8
+        
+        long[] buckets = stats.getSizeBuckets();
+        Assertions.assertThat(buckets[0]).isEqualTo(1); // size 1
+        Assertions.assertThat(buckets[1]).isEqualTo(1); // size 2
+        Assertions.assertThat(buckets[2]).isEqualTo(1); // size 4
+        Assertions.assertThat(buckets[3]).isEqualTo(1); // size 8
+        
+        // All other buckets should be zero
+        for (int i = 4; i < Integer.SIZE; i++) {
+            Assertions.assertThat(buckets[i]).isZero();
+        }
+    }
+
+    @Test
+    public void sizeBucketsZeroSizeHandling() {
+        SizeStatisticsResults stats = new SizeStatisticsResults();
+        
+        // Zero-size key-value pair should not be counted in buckets
+        stats.updateStatistics(new KeyValue("".getBytes(), "".getBytes()));
+        
+        long[] buckets = stats.getSizeBuckets();
+        Assertions.assertThat(buckets).containsOnly(0L);
+        Assertions.assertThat(stats.getKeyCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void proportionCalculation() {
+        SizeStatisticsResults stats = new SizeStatisticsResults();
+        
+        // Add entries with known sizes to specific buckets
+        for (int i = 0; i < 10; i++) {
+            stats.updateStatistics(new KeyValue("a".getBytes(), Strings.repeat("a", i).getBytes()));
+        }
+
+        // Total: 30 entries, 10 in bucket 1, 20 in bucket 2
+        Assertions.assertThat(stats.getKeyCount()).isEqualTo(10);
+        
+        // Test various proportions
+        double p10 = stats.getProportion(0.1); // 10% = 3 entries, should be in bucket 1
+        double p50 = stats.getProportion(0.5); // 50% = 15 entries, should be in bucket 2
+        double p90 = stats.getProportion(0.9); // 90% = 27 entries, should be in bucket 2
+        
+        Assertions.assertThat(p10).isGreaterThan(1.0).isLessThanOrEqualTo(2.0);
+        Assertions.assertThat(p50).isGreaterThan(5.0).isLessThanOrEqualTo(6.0);
+        Assertions.assertThat(p90).isGreaterThan(9.0); // bucket 2 range
+    }
+
+    @Test
+    public void medianP90P95() {
+        SizeStatisticsResults stats = new SizeStatisticsResults();
+        
+        // Add uniform distribution
+        for (int i = 0; i < 100; i++) {
+            stats.updateStatistics(new KeyValue("ab".getBytes(), "cd".getBytes())); // size 4
+        }
+        
+        double median = stats.getMedian();
+        double p90 = stats.getP90();
+        double p95 = stats.getP95();
+        
+        Assertions.assertThat(median).isEqualTo(stats.getProportion(0.5));
+        Assertions.assertThat(p90).isEqualTo(stats.getProportion(0.9));
+        Assertions.assertThat(p95).isEqualTo(stats.getProportion(0.95));
```

Comment:
These tests seem a tad tautological, as it's basically showing (at best) that median is syntactic sugar for `getProportion(0.5)`. Should there be a variant that asserts on the absolute values returned here? We may also need to test this out on a non-uniform distribution to get signal here

Reply:
These were added to cover a test gap but indeed, all these methods are doing is syntactic sugar. I am not sure how we can provide more meaningful tests for them...

I added one more random test (with the hopes it will not become flaky), to test the results of proportion calculation.

Reply:
Deleted that test and added more assertions to proportionCalculation that has fixed values. Hope that makes more sense